src/metabase/models/collection.clj

```diff
@@ -333,25 +333,40 @@
       (and first-name last-name) (trs "{0} {1}''s Personal Collection" first-name last-name)
       :else                      (trs "{0}''s Personal Collection" (or first-name last-name email)))))
 
+(mu/defn user->personal-collection-names :- ms/Map
+  "Come up with a nice name for the Personal Collection for the passed `user-or-ids`.
+  Returns a map of user-id -> name"
+  [user-or-ids user-or-site]
+  (let [ids (filter some? (map u/the-id user-or-ids))
```

Comment:
just for your interest, not a required change - toucan has tons of cool functions like `t2/select-pk->fn`, which does exactly what you want here:

```
(mu/defn user->personal-collection-names :- ms/Map
  "Come up with a nice name for the Personal Collection for the passed `user-or-ids`.
  Returns a map of user-id -> name"
  [user-or-ids user-or-site]
  (when-let [ids (seq (filter some? (map u/the-id user-or-ids)))]
    (t2/select-pk->fn #(format-personal-collection-name (:first_name %) (:last_name %) (:email %) user-or-site)
                      [:model/User :first_name :last_name :email :id]
                      :id [:in ids])))
```